### PR TITLE
Refs [TASK][34935] Handle event Genre screen

### PR DESCRIPTION
**Pull request by manhhung23399**
## Related Tickets
- [#Task 34935: Handle event Genre screen](https://edu-redmine.sun-asterisk.vn/issues/34935)
## WHAT
- Create request to themoviedb api to get data(Movie by genres) which parse to models then pour data from remote data
source into View
- Handle click Back button
## Evidence (Screenshot or Video)
image

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/moviews/data/MovieDataSource.kt [hunk 1]
```diff
@@ -9,11 +9,13 @@ interface MovieDataSource {
...
     }
-    interface Remote{
+
+    interface Remote {
         fun getTrendingMovie(callback: OnLoadDataCallback<MutableList<Movie>>)
         fun getUpcomingMovie(callback: OnLoadDataCallback<MutableList<Movie>>)
         fun getNowPlayingMovie(callback: OnLoadDataCallback<MutableList<Movie>>)
         fun getPopularMovie(callback: OnLoadDataCallback<MutableList<Movie>>)
         fun getTopSearchMovie(callback: OnLoadDataCallback<MutableList<Movie>>)
+        fun getMovieByGenre(idGenre: Int, callback: OnLoadDataCallback<MutableList<Movie>>)
```
Comment (tuannm-1276) [inline app/src/main/java/com/example/moviews/data/MovieDataSource.kt]:
```suggestion
        fun getMovieByGenres(idGenre: Int, callback: OnLoadDataCallback<MutableList<Movie>>)
```
hình như mấy chỗ trên nữa

File: app/src/main/res/layout/item_movie_list.xml [hunk 2]
```diff
@@ -7,15 +7,15 @@
...
+        android:id="@+id/imageMovieList"
         android:layout_width="@dimen/dp_180"
         android:layout_height="@dimen/dp_120"
         android:scaleType="fitXY"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toTopOf="parent" />
 
     <TextView
-        android:id="@+id/textTitleMovieSearch"
+        android:id="@+id/textTitleMovieList"
```
Comment (tuannm-1276) [inline app/src/main/res/layout/item_movie_list.xml]:
```suggestion
        android:id="@+id/textTitle"
```

File: app/src/main/res/layout/item_movie_list.xml [hunk 3]
```diff
@@ -7,15 +7,15 @@
     android:background="@color/color_dark_gray_light2">
 
     <ImageView
-        android:id="@+id/imageMovieSearch"
+        android:id="@+id/imageMovieList"
```
Comment (tuannm-1276) [inline app/src/main/res/layout/item_movie_list.xml]:
```suggestion
        android:id="@+id/imageMovie"
```

Comment (tuannm-1276) [reply app/src/main/res/layout/item_movie_list.xml]: ngoài item_movie_list còn có item_movie hả em

Comment (manhhung23399) [reply app/src/main/res/layout/item_movie_list.xml]: có a ạ, item_movie là của màn home với cast_detail

**Pull request merged**